runtime/Cpp/runtime/src/misc/IntervalSet.cpp

```diff
@@ -46,6 +46,21 @@ IntervalSet::IntervalSet(int n, ...) : IntervalSet() {
   }
 }
 
+IntervalSet::~IntervalSet()
+{
+}
+
+IntervalSet& IntervalSet::operator=(const IntervalSet& other)
+{
+  if (_readonly) {
+    throw IllegalStateException("can't alter read only IntervalSet");
+  }
```

Comment:
Hmm. That implies that the meaning of `_readonly` is different to my usage in the concurrency patch.

However, is that really true? This code detected the double assignment in `ATN::nextStates(ATNState*)`, and C++ operator= is not the same as Java reference assignment.

The possibilities I see:

1. Copy `_readonly` in the operator=(), have a different atomic field for use by ATN to detect races.
2. Leave the behaviour as in this patch, still add a different atomic field for use by ATN.
3. Leave the patch as-is.

There are probably other options I've missed. What do you think?